weave/trace_server/calls_query_builder.py

```diff
@@ -893,6 +926,80 @@ def process_calls_filter_to_conditions(
     return conditions
 
 
+def make_call_parts_predicate_filters_sql(
+    pb: ParamBuilder, table_alias: str, heavy_conditions: list[Condition]
+) -> str:
```

Comment:
I would add the comment here that the condition returned MUST account for unmatched parts and NEVER filter out more than the heavy filter would